edgelet/edgelet-http/src/certificate_manager.rs

```diff
@@ -0,0 +1,175 @@
+use std::str;
+use std::sync::{Arc, RwLock};
+
+use edgelet_core::crypto::{Certificate, CreateCertificate};
+use edgelet_core::{CertificateIssuer, CertificateProperties, CertificateType};
+use failure::ResultExt;
+
+pub use crate::error::{Error, ErrorKind};
+
+const IOTEDGED_VALIDITY: u64 = 7_776_000; // 90 days
+const IOTEDGED_TLS_COMMONNAME: &str = "iotedge tls";
+
+#[derive(Clone)]
+pub struct CertificateManager<C: CreateCertificate + Clone> {
+    certificate: Arc<RwLock<String>>,
```

Comment:
Instead of treating `""` as the sentinel "non-existent" value, seems like this ought to be `Arc<RwLock<Option<String>>>` ?